app/Livewire/Components/ChangeLocale.php

```diff
@@ -19,22 +17,23 @@ public function mount(): void
         $this->selectedLang = app()->getLocale();
     }
 
-    public function changeLang(string $lang): RedirectResponse|Redirector
+    public function changeLang(string $lang): void
     {
         $user = Auth::user();
+
         if ($user) {
             $settings = $user->settings;
-            $settings['default_lang'] = $lang;
+            $settings['locale'] = $lang;
             $user->settings = $settings;
             $user->save();
         }
-        app()->setLocale($lang);
 
-        return redirect()->to(url()->current());
+        app()->setLocale($lang);
+        $this->dispatch('localeChanged');
```

Comment:
Quelle erreur sort quand tu utilises un redirect ?

Reply:
![livewire-error](https://github.com/user-attachments/assets/2983dae7-f954-4a82-bf86-295247726143)

Reply:
Et c'est la même erreur quand tu fais `$this->redirect('/')` ?

Reply:
oui que ce soit une redirection laravel ou une redirection livewire le comportement est pareil